doc/reference/reference_lua/box_ctl/on_recovery_state.rst

```diff
@@ -0,0 +1,49 @@
+..  _box_ctl-on_recovery_state:
+
+===============================================================================
+box.ctl.on_recovery_state()
+===============================================================================
+
+..  module:: box.ctl
+
+..  function:: on_recovery_state(trigger-function)
+
+    **Since:** :doc:`2.11.0 </release/2.11.0>`
+
+    Create a :ref:`trigger <triggers>` executed on different stages of a node :ref:`recovery <internals-recovery_process>` or initial configuration.
+    Note that you need to set the ``box.ctl.on_recovery_state`` trigger before the initial :ref:`box.cfg <box_introspection-box_cfg>` call.
+
+    :param function     trigger-function: a trigger function
+
+    :return: ``nil`` or a function pointer
+
+    A registered trigger function is run on each of the supported recovery
+    state and receives the state name as a parameter:
+
+    *   ``snapshot_recovered``: the node has recovered the snapshot files.
+    *   ``wal_recovered``: the node has recovered the WAL files.
+    *   ``indexes_built``: the node has built secondary indexes for memtx spaces.
+        This stage might come before any actual data is recovered, meaning that the
+        indexes are available right after the first tuple is recovered.
+    *   ``synced``: the node has synced with enough remote peers.
+        This means that the node changes the state from :ref:`orphan <internals-replication-orphan_status>` to ``running``.
+
+    All these states are passed during initial ``box.cfg`` call when recovering
```

Comment:
`synced` state might happen **after** the initial `box.cfg` call exits: for example, if the user has `replication_sync_timeout = 0`, the node will exit `box.cfg` without reaching `synced`, but staying orphan.

Once enough remote peers are synced with, the `synced` recovery state will be reached.

Reply:
thanks! Added this information to the description.